src/interpret/heaptrack_interpret.cpp

```diff
@@ -304,7 +304,36 @@ struct AccumulatedTraceData
         dwfl_end(m_dwfl);
     }
 
-    /// find a file in the sysroot or extra path
+    /// find a file in the sysroot or extra path and add it to the cache
+    const std::string& resolveAndCacheFile(const std::string& fileName)
+    {
+        if (!m_extraPaths.empty()) {
+            // look for the filename, checking both if it's there using the defined path
```

Comment:
imo, we should do first one loop that finds the files using one approach (e.g. the full path, like you propose), and then do a second loop as a fallback with the other approach

the mixed approach of your code might behave in funky ways when there are multiple extra paths, no?

Reply:
the way your new version (and my adapted one) handle it is fine - at least we'll always prefer one location over the other now, independent of how we traverse the folder structure.